docs/keys.md

```diff
@@ -1,10 +1,8 @@
 # Keys
 
-Every time your application is rendered, a [virtual node](/docs/virtual-node.md) tree is created from scratch.
+Every time your application is rendered, a [virtual node](/docs/virtual-node.md) tree is created from scratch. This means that between renders, there is no definite mapping from a virtual node to a particular DOM element.
 
-Keys help identify which nodes were added, changed or removed in a list when comparing the old and new tree.
-
-Use keys to identify items in a list that can be re-ordered or where items can be added to / removed from.
+However, nodes with the same `key` accross renders will be interpreted as representing a particular DOM element, allowing you to force dom elements to be reordered, created or destroyed, rather than repurposed. This is sometimes important in order to avoid resetting data attached to DOM elements by the browser (for example the binary image data associated with an `<img src="..." />`), or in [lifecycle events](/docs/lifecycle-events.md). 
```

Comment:
Can you work a bit more on this sentence? It's quite verbose.

Also a couple of petpeeves:

- dom -> DOM (thought it would be nicer if you could find a way to not have to mention DOM so many times).
- lifecycle events -> VirtualDOM Events. 
  - https://github.com/hyperapp/hyperapp/blob/master/docs/vdom-events.md